tezos-encoding/src/nom.rs

```diff
@@ -572,6 +573,56 @@ where
     }
 }
 
+#[derive(Debug, PartialEq)]
+/// Lazy, memoized deserialization of a dynamic block.
+pub struct LazyCell<'a, T>
```

Comment:
this should probably exist in `types.rs` instead? (Like `Bytes`, `SizedBytes` etc)